File: ios/Sources/Ratakierros/Models/AuthResponse.swift
import Foundation

struct AuthResponse: Codable {
    let token: String
    let userId: Int64
    let displayName: String

    enum CodingKeys: String, CodingKey {
        case token
        case userId = "user_id"
        case displayName = "display_name"
    }
}

struct LoginRequest: Codable {
    let email: String
    let password: String
}

struct RegisterRequest: Codable {
    let email: String
    let displayName: String
    let password: String

    enum CodingKeys: String, CodingKey {
        case email
        case displayName = "display_name"
        case password
    }
}

struct APIErrorPayload: Codable {
    let error: String?
}

File: android/app/src/main/kotlin/fi/ratakierros/model/AuthResponse.kt
package fi.ratakierros.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AuthResponse(
    val token: String,
    @SerialName("user_id") val userId: Long,
    @SerialName("display_name") val displayName: String,
)

@Serializable
data class LoginRequest(val email: String, val password: String)

@Serializable
data class RegisterRequest(
    val email: String,
    @SerialName("display_name") val displayName: String,
    val password: String,
)

@Serializable
data class ApiErrorPayload(val error: String? = null)
